custom_model_runner/datarobot_drum/drum/utils.py

```diff
@@ -348,8 +343,18 @@ def read_structured_input_data_as_df(binary_data, mimetype, sparse_colnames=None
                 df.fillna(value=np.nan, inplace=True)
 
                 return df
-            else:
-                return pd.read_csv(io.BytesIO(binary_data))
+            else:  # CSV format
+                df = pd.read_csv(io.BytesIO(binary_data))
+
+                # If the DataFrame only contains a single column, treat blank lines as NANs
+                if df.shape[1] == 1:
```

Comment:
is it possible for the shape to be 1d?

Reply:
I don't think dataframes can be 1d

Reply:
(this will always return a dataframe)

Reply:
yup you're good: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.shape.html